### PR TITLE
List viewed/new status by instrument

### LINKED ISSUE #1073: new/viewed table at 'Archived Images' page
In order to facilitate the internal NIRSpec keeping track of the PIDs and images that have been reviewed, we would like to be able to export a table with the following information (perhaps at the 'Archived Images' page):

Filename  |   EXP_TYPE   |   Filter   |    Grating   |   PATT_NUM   |   Viewed

JW01180027001_02101_00001_NRS1_CAL  |   NRS_MSATA   |   CLEAR    |     PRISM    |    1    |     yes

### LINKED ISSUE #1118: Set default sorting button labels
At the moment, when an instrument's Archive page is loaded, the proposals are ordered by Descending proposal number. However, the sorting button on the right side is labeled 'Ascending'. We should set the default button value to Descending in order to avoid confusion.

### LINKED ISSUE #1154: Return a list of viewed/new files
NIRSpec request:

Add the ability to create a list of new/viewed files for a given archive/proposal/observation.

@penaguerrero any other details on this? Is this something we should add to our API, so that it will give you a json list of files that can be downloaded? (Note that for our API you have to visit the link in a browser. We can't access API pages from a script due to the SSO layer between JWQL and the outside world.)

### PR DESCRIPTION
**Pull request by melanieclarke**
Adds a button to the instrument archive pages to download a CSV report containing the new/viewed status by root filename.  Additional useful observation descriptors, configured by instrument, are also added to the report.  These may be any field defined in the Django models (RootFileInfo, Observation, or Proposal).

If accessed via the archive page button, the report will respect the current filter and sort settings on the page. The report can also be downloaded directly at {instrument}/report, with optional filtering and sorting passed as query parameters, e.g. {instrument}/report?sort_as=descending&look=viewed.

For a JSON report of the same data, there is an API endpoint at api/{instrument}/looks.

### CODE REVIEW

File: jwql/tests/test_data_containers.py [hunk 1]
```diff
@@ -115,6 +170,63 @@ def test_get_instrument_proposals():
     assert len(proposals) > 0
 
 
+@pytest.mark.skipif(ON_GITHUB_ACTIONS, reason='Requires access to central storage.')
```
Comment (mfixstsci) [inline jwql/tests/test_data_containers.py:224]: This looks great, love all of these new tests, thank you!

File: jwql/utils/constants.py [hunk 2]
```diff
@@ -468,6 +469,17 @@ class QUERY_CONFIG_KEYS:
...
+
+REPORT_KEYS_PER_INSTRUMENT = {'fgs': ['proposal', 'obsnum', 'exptypes',
+                                      'obsstart', 'obsend', 'viewed'],
+                              'miri': ['proposal', 'obsnum', 'exptypes',
+                                       'obsstart', 'obsend', 'viewed'],
+                              'nircam': ['proposal', 'obsnum', 'exptypes',
+                                         'obsstart', 'obsend', 'viewed'],
+                              'niriss': ['proposal', 'obsnum', 'exptypes',
+                                         'obsstart', 'obsend', 'viewed'],
+                              'nirspec': ['exptypes', 'viewed']}
```
Comment (mfixstsci) [inline jwql/utils/constants.py]: This looks great, were these keys chosen by the instrument teams? Any reason we didn't include proposal, obsnum etc?

File: jwql/website/apps/jwql/data_containers.py [hunk 3]
```diff
@@ -532,7 +536,7 @@ def get_filenames_by_instrument(instrument, proposal, observation_id=None, restr
         filenames in this file will be used rather than calling mask_query_filenames_by_instrument.
         This can save a significant amount of time when the number of files is large.
     query_response : dict
-        Dictionary with "data" key ontaining a list of filenames. This is assumed to
+        Dictionary with "data" key containing a list of filenames. This is assumed to
```
Comment (mfixstsci) [inline jwql/website/apps/jwql/data_containers.py:716]: Thanks 😅

File: jwql/website/apps/jwql/data_containers.py [hunk 4]
```diff
@@ -666,18 +734,7 @@ def get_filenames_by_rootname(rootname):
...
-
-    filenames = glob.glob(os.path.join(FILESYSTEM_DIR, 'public', proposal_dir, observation_dir, '{}*'.format(rootname)))
-    filenames.extend(glob.glob(os.path.join(FILESYSTEM_DIR, 'proprietary', proposal_dir, observation_dir, '{}*'.format(rootname))))
-
-    # Certain suffixes are always ignored
-    filenames = [filename for filename in filenames if os.path.splitext(filename)[0].split('_')[-1] not in IGNORED_SUFFIXES]
-    filenames = sorted([os.path.basename(filename) for filename in filenames])
-
-    return filenames
+    return get_filesystem_filenames(rootname=rootname)
```
Comment (mfixstsci) [inline jwql/website/apps/jwql/data_containers.py:963]: Two for one special, love to see it.

**Review by mfixstsci — APPROVED** [review]
Hey @melanieclarke everything here looks good to me. Thank you for the tests and doc updates! I wanted to ask about the loading times, I know for other download tools in the app when you click the download button, you are presented with a dialog box and can select where you'd like to save your tables or data. I am curious if we wanted to do something to show download progress here as well just so users don't click around too much. @bhilbert4 or @BradleySappington have you tested these new changes?

File: jwql/utils/constants.py (hunk 2 above)
Comment (melanieclarke) [reply jwql/utils/constants.py]: For this one, we still need to get instrument team feedback.  

For now, for non-NIRSpec keys, I just included everything currently available.  

For NIRSpec, they requested exp_type, filter, grating, patt_num, and viewed, of which only exp_type and viewed is currently available.  I'll add in the others when we get them in the models.

**Comment by melanieclarke** [comment]
> Hey @melanieclarke everything here looks good to me. Thank you for the tests and doc updates! I wanted to ask about the loading times, I know for other download tools in the app when you click the download button, you are presented with a dialog box and can select where you'd like to save your tables or data. I am curious if we wanted to do something to show download progress here as well just so users don't click around too much. @bhilbert4 or @BradleySappington have you tested these new changes?

For the download button, I was modeling it on the download button for the JWQLDB tab, which just downloads the CSV file when clicked.  Is there a better model somewhere else in the app?

File: jwql/utils/constants.py (hunk 2 above)
Comment (melanieclarke) [reply jwql/utils/constants.py]: Update: NIRSpec keys are now as requested, using the new fields from the models.

For all other instruments, the keys returned are proposal, exp_type, expstart, filter, aperture, detector, and subarray.  For NIRCam and NIRISS, I added pupil at Bryan's request.  These keys can be easily configured later as needed.

**Review by BradleySappington — APPROVED** [review]
Looks great to me!

**Pull request merged**